synapse/links/ssl.py

```diff
@@ -11,6 +11,38 @@
 
 logger = logging.getLogger(__name__)
 
+class SslSocket(s_socket.Socket):
+
+    def __init__(self, sock, **info):
+        s_socket.Socket.__init__(self, sock, **info)
+
+    def send(self, byts):
+        try:
+            return self.sock.send(byts)
+        except ssl.SSLWantWriteError as e:
+            return 0
+
+    def accept(self):
```

Comment:
doing this means Socket subclasses have to be aware of changes to the base Socket.accept implementation

Reply:
Could this be implemented in the base class with another use of self.__class__?